examples/flutter_search/lib/main.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import './bloc/bloc.dart';
+import './models/entities.dart';
+import './github_service/github_service.dart';
+
+void main() => runApp(SearchApp());
+
+class SearchApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Search for articles',
+      home: Scaffold(
+        appBar: AppBar(title: Text("Enter a search term")),
+        body: SearchForm(),
+      ),
+    );
+  }
+}
+
+// Define a Custom Form Widget
+class SearchForm extends StatefulWidget {
+  @override
+  _SearchFormState createState() => _SearchFormState();
+}
+
+// Define a corresponding State class. This class will hold the data related to
+// our Form.
+class _SearchFormState extends State<SearchForm> {
+  // Create a text controller. We will use it to retrieve the current value
+  // of the TextField!
+  final _textController = TextEditingController();
+
+  final GithubSearchBloc _githubSearchBloc = GithubSearchBloc(
+      GithubService(GithubCache(), GithubClient(http.Client())));
+
+  @override
+  void initState() {
+    _githubSearchBloc.dispatch(TextChanged(text: ""));
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    // Clean up the controller when the Widget is removed from the Widget tree
+    // This also removes the _printLatestValue listener
+    _githubSearchBloc.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<GithubSearchEvent, GithubSearchState>(
+      bloc: _githubSearchBloc,
+      builder: (BuildContext context, GithubSearchState state) {
+        return Column(
+          children: <Widget>[
+            Padding(
+              padding: EdgeInsets.only(top: 20.0),
+            ),
+            TextField(
+              controller: _textController,
+              onChanged: (text) {
+                _githubSearchBloc.dispatch(TextChanged(text: text));
+              },
+              decoration: InputDecoration(
+                  border: InputBorder.none,
+                  hintText: 'Please enter a search term'),
+            ),
+            _buildResult(state)
```

Comment:
it's actually a bad pattern to write functions that return widgets because then Flutter doesn't know about the widget tree and can't perform optimizations. https://stackoverflow.com/questions/53234825/what-is-the-difference-between-functions-and-classes-to-create-widgets/53234826#53234826

As a result, I'd recommend moving the `_buildResult` logic directly into the build method.